controller/recurring_job_controller.go

```diff
@@ -220,6 +224,53 @@ func (control *RecurringJobController) syncRecurringJob(key string) (err error)
 	return nil
 }
 
+func (control *RecurringJobController) cleanupVolumeRecurringJob(recurringJob *longhorn.RecurringJob) error {
+	// Check if each group of the recurring job contains other recurring jobs.
+	// If No, it means the recurring job is the last job of the group then
+	// Longhorn will clean up this group labels for all volumes.
+	checkRecurringJobs, err := control.ds.ListRecurringJobs()
+	if err != nil {
+		return err
+	}
+	rmGroups := []string{}
+	for _, group := range recurringJob.Spec.Groups {
+		inUse := false
+		for _, checkRecurringJob := range checkRecurringJobs {
+			if checkRecurringJob.Name == recurringJob.Name {
+				continue
+			}
+			if util.Contains(checkRecurringJob.Spec.Groups, group) {
+				inUse = true
+				break
+			}
+		}
+		if !inUse {
+			rmGroups = append(rmGroups, group)
+		}
+	}
+
+	// delete volume labels
+	volumes, err := control.ds.ListVolumesRO()
```

Comment:
When you decide to mutate the objects in this list result later, please do not use `RO`. Or you can rerun the get function before the mutator.

Reply:
Fixed.